sa/sa.go

```diff
@@ -1281,8 +1301,26 @@ func (ssa *SQLStorageAuthority) getNewIssuancesByFQDNSet(
 // FQDNSetExists returns a bool indicating if one or more FQDN sets |names|
 // exists in the database
 func (ssa *SQLStorageAuthority) FQDNSetExists(ctx context.Context, names []string) (bool, error) {
+	tx, err := ssa.dbMap.Begin()
```

Comment:
Putting this in a transaction in order to satisfy `AddCertificate` needing to use `checkFQDNSetExists` in a context feels a bit wrong.

Perhaps `checkFQDNSetExists` could instead take a argument `func(interface{}, string, ...interface{}) error` for the selector method instead of a transaction obj and then `AddCertificate` could pass in `tx.SelectOne` and `FQDNSetExists` could pass in `ssa.dbMap.WithContext(ctx).SelectOne`?

Reply:
Good call, I was being lazy. Fixed in 07bd54b